Added mock context decorator to storybook

rasmuswoelk opened the pull request:
I noticed that some of the components failed to render in storybook after the addition of different context dependable features - such as the intl-context.

![image](https://user-images.githubusercontent.com/640976/40291621-9ad37eba-5cf8-11e8-8068-297fd6da749d.png)

To fix this I have added a mock context decorator which mimics the context from stripes-core to prevent certain component examples from failing.

This is probably a temporary fix but it gets the job done.

id-jenkins commented:
<html><body><pre>
yarn run v1.6.0
$ eslint lib && stylelint "lib/**/*.css"

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/Dropdown/Dropdown.js
  52:1  error  componentWillUpdate is deprecated since React 16.3.0, use UNSAFE_componentWillUpdate instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillupdate  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/Layer/Layer.js
  29:1  error  componentWillMount is deprecated since React 16.3.0, use UNSAFE_componentWillMount instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillmount  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/MultiColumnList/MCLRenderer.js
  71:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/Selection/SelectList.js
  49:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/Selection/SingleSelect.js
  65:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/SRStatus/SRStatus.js
  4:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/structures/AddressFieldGroup/AddressList/AddressList.js
  102:1  error  componentWillMount is deprecated since React 16.3.0, use UNSAFE_componentWillMount instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillmount  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/structures/EditableList/EditableListForm.js
  136:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/Timepicker/TimeDropdown.js
  37:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/Timepicker/Timepicker.js
  64:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

✖ 10 problems (10 errors, 0 warnings)

info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
</pre><body></html>

id-jenkins commented:
<html><body><pre>
yarn run v1.6.0
$ eslint lib && stylelint "lib/**/*.css"

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/Dropdown/Dropdown.js
  52:1  error  componentWillUpdate is deprecated since React 16.3.0, use UNSAFE_componentWillUpdate instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillupdate  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/Layer/Layer.js
  29:1  error  componentWillMount is deprecated since React 16.3.0, use UNSAFE_componentWillMount instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillmount  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/MultiColumnList/MCLRenderer.js
  71:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/Selection/SelectList.js
  49:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/Selection/SingleSelect.js
  65:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/SRStatus/SRStatus.js
  4:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/structures/AddressFieldGroup/AddressList/AddressList.js
  102:1  error  componentWillMount is deprecated since React 16.3.0, use UNSAFE_componentWillMount instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillmount  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/structures/EditableList/EditableListForm.js
  136:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/Timepicker/TimeDropdown.js
  37:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/Timepicker/Timepicker.js
  64:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

✖ 10 problems (10 errors, 0 warnings)

info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
</pre><body></html>

rasmuswoelk commented:
Hmm.. I guess it makes sense to have these functions come from stripes-core since they will be used anywhere. 

However, this might be another reason for moving to a more customized setup instead of using storybook?

cherewaty commented:
Yesterday I was reading about context on the `react-intl` docs and they mentioned this: https://github.com/truffls/storybook-addon-intl

Would that storybook add-on simplify this?

rasmuswoelk commented:
@cherewaty Brilliant. I have updated the config to use storybook-addon-intl instead. It seems to be working nicely.

I removed my mock context for now but leaving the file there - just in case we need it to mock some other context down the line.

id-jenkins commented:
<html><body><pre>
yarn run v1.6.0
$ eslint lib && stylelint "lib/**/*.css"

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/Layer/Layer.js
  29:1  error  componentWillMount is deprecated since React 16.3.0, use UNSAFE_componentWillMount instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillmount  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/MultiColumnList/MCLRenderer.js
  71:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/Selection/SelectList.js
  49:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/Selection/SingleSelect.js
  65:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/SRStatus/SRStatus.js
  4:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/structures/AddressFieldGroup/AddressList/AddressList.js
  102:1  error  componentWillMount is deprecated since React 16.3.0, use UNSAFE_componentWillMount instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillmount  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/structures/EditableList/EditableListForm.js
  136:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/Timepicker/TimeDropdown.js
  37:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/Timepicker/Timepicker.js
  64:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

✖ 9 problems (9 errors, 0 warnings)

info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
</pre><body></html>

id-jenkins commented:
<html><body><pre>
yarn run v1.7.0
$ eslint lib && stylelint "lib/**/*.css"

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/Layer/Layer.js
  29:1  error  componentWillMount is deprecated since React 16.3.0, use UNSAFE_componentWillMount instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillmount  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/MultiColumnList/MCLRenderer.js
  71:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/Selection/SelectList.js
  49:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/Selection/SingleSelect.js
  65:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/SRStatus/SRStatus.js
  4:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/structures/AddressFieldGroup/AddressList/AddressList.js
  102:1  error  componentWillMount is deprecated since React 16.3.0, use UNSAFE_componentWillMount instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillmount  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/structures/EditableList/EditableListForm.js
  136:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/Timepicker/TimeDropdown.js
  37:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

/home/jenkins/workspace/folio-org/stripes-components/PR-380/project/lib/Timepicker/Timepicker.js
  64:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated

✖ 9 problems (9 errors, 0 warnings)

info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
</pre><body></html>
